tests/unit/transformers/test_base.py

```diff
@@ -1,10 +1,374 @@
+from unittest.mock import patch
+
 import pandas as pd
+import pytest
 
 from rdt.transformers.base import BaseTransformer
 
 
 class TestBaseTransformer:
 
+    def test_get_subclasses(self):
+        """Test the ``get_subclasses`` method.
+
+        Setup:
+            -
+
+        Input:
+            -
+
+        Expected behavior:
+            -
+        """
+        pass  # TODO: not sure how to test this one.
+
+    def test_get_input_type(self):
+        """Test the ``get_input_type`` method.
+
+        Setup:
+            - mock the ``BaseTransformer``'s ``INPUT_TYPE`` attribute.
+
+        Output:
+            - the string stored in the ``INPUT_TYPE`` attribute.
+        """
+        with patch.object(BaseTransformer, 'INPUT_TYPE', 'categorical'):
+            # Run
+            input_type = BaseTransformer.get_input_type()
+
+            # Assert
+            expected = 'categorical'
+            assert input_type == expected
+
+    def test__add_prefix_none(self):
+        """Test the ``_add_prefix`` method when passed a ``None``.
+
+        The method should return ``None`` when ``None`` is passed as an argument.
+
+        Input:
+            - a None value.
+
+        Output:
+            - a None value.
+        """
+        # Setup
+        dictionary = None
+        base_transformer = BaseTransformer()
+
+        # Run
+        output = base_transformer._add_prefix(dictionary)
+
+        # Assert
+        expected = None
+        assert output == expected
+
+    def test__add_prefix_dictionary(self):
+        """Test the ``_add_prefix`` method when passed a dictionary.
+
+        Setup:
+            - set the ``column_prefix`` of the ``BaseTransformer`` to ``'colulumn_name'``.
+
+        Input:
+            - a dictionary of strings to strings.
+
+        Output:
+            - the input dictionary with ``column_prefix`` added to the beginning of the keys.
+        """
+        # Setup
+        dictionary = {
+            'day': 'numerical',
+            'month': 'categorical',
+            'year': 'numerical'
+        }
+        transformer = BaseTransformer()
+        transformer.column_prefix = 'column_name'
+
+        # Run
+        output = transformer._add_prefix(dictionary)
+
+        # Assert
+        expected = {
+            'column_name.day': 'numerical',
+            'column_name.month': 'categorical',
+            'column_name.year': 'numerical'
+        }
+        assert output == expected
+
+    def test_get_output_types(self):
+        """Test the ``get_output_types`` method.
+
+        Setup:
+            - set the ``column_prefix`` of the ``BaseTransformer`` to ``'colulumn_name'``.
```

Comment:
Let's use a subclass like in the `get_input_type` above.
Notice that this particular implementation is changing the value of the `OUTPUT_TYPES` of the base class, potentially affecting other tests that depend on it.